quartz/worker.ts

```diff
@@ -3,23 +3,46 @@ sourceMapSupport.install(options)
 import cfg from "../quartz.config"
 import { Argv, BuildCtx } from "./util/ctx"
 import { FilePath, FullSlug } from "./util/path"
-import { createFileParser, createProcessor } from "./processors/parse"
+import {
+  createFileParser,
+  createHtmlProcessor,
+  createMarkdownParser,
+  createMdProcessor,
+} from "./processors/parse"
 import { options } from "./util/sourcemap"
+import { MarkdownContent, ProcessedContent } from "./plugins/vfile"
 
 // only called from worker thread
-export async function parseFiles(
+export async function parseMarkdown(
   buildId: string,
   argv: Argv,
   fps: FilePath[],
+): Promise<[MarkdownContent[], FullSlug[]]> {
+  // this is a hack
+  // we assume markdown parsers can add to `allSlugs`,
+  // but don't actually use them
```

Comment:
This is a big assumption imho.

I think there is no preventing users from using allSlugs in markdown in this case?

Reply:
Well the thing is if some markdown transformer adds to allSlugs and some other uses allSlugs - there will always be ordering issues similar to what I'm trying to prevent with html

So basically it makes little sense to do that.

This could kind of be fixed by assigning priorities to transformers and then doing multiple stages for all files (again literally the same thing as I'm doing with html, but multiple times), but that's a whole lot of work for no gain as we don't use that anyway.

And what I just said also makes no sense with workers as we actually need to have a concurrently shared list that different workers need to add to and read from(!) which I've no idea how to do (imagine having normal threads and synchronization tools in node) - and again, a whole lotta complexity for "correctness", or this little (but very ugly) hack we have here 😅